airflow/models.py

```diff
@@ -2412,15 +2414,28 @@ def __deepcopy__(self, memo):
         memo[id(self)] = result
 
         for k, v in list(self.__dict__.items()):
-            if k not in ('user_defined_macros', 'user_defined_filters', 'params'):
+            if k not in ('user_defined_macros', 'user_defined_filters',
+                         'params', '_log'):
                 setattr(result, k, copy.deepcopy(v, memo))
         result.params = self.params
         if hasattr(self, 'user_defined_macros'):
             result.user_defined_macros = self.user_defined_macros
         if hasattr(self, 'user_defined_filters'):
             result.user_defined_filters = self.user_defined_filters
+        if hasattr(self, '_log'):
+            result._log = self._log
         return result
 
+    def __getstate__(self):
+        state = dict(self.__dict__)
```

Comment:
NIT: may be safer to do a check `if hasattr(state, '_log')` here in case it doesn't exist and throw a KeyError. i don't think the error is possible right now in code but just in case code change in the future?